### DISCUSSION TITLE
Apply branch schema changes to any other branch, not just 'main'

### DISCUSSION THREAD

**Discussion by grempe** [planetscale/discussion#405] [Feature Requests] [answered]
Trying to map planetscale branches to a dev -> staging -> prod flow where I'd like to be able to push schema changes to any development branch and choose *any* other branch (not just the single production branch) as the target of that branch merge.

So for example, dev could be merged to staging, and later staging could be merged to production.

I think this would go a long way to solving e.g the Vercel preview branch issues and would add a massive new flexibility in how developers use Planetscale.

It would also allow stability in the connection strings used in the staging and production configurations.

**Comment by mscoutermarsh** [comment, marked as answer]
Hi @grempe,

You could make your `staging` a production branch as well. This would enable using deploy requests to merge schema changes into it.

The "base" branch in a deploy request has to be production because production branches have "ddl protection" enabled. This stops schema changes from being run by anything other than deploy requests.

**The flow would work like this:**
You would create two production branches, e.g. main and staging. You then work on your new feature in an ephemeral development branch (20220302-new-feature in this example). Once work is done, you would deploy from 20220302-new-feature to staging and then from staging to main.

**Reply by grempe** [reply]
This is the answer.  The key for me was understanding that:

1) having both 'staging' and 'production' branches be production branches.
2) hanging the ephemeral dev branches off of 'staging', 'production' needs no branches.
3) deploying branch changes from ephemeral dev branches to 'staging' first and then later creating a branch deploy to promote all changes from staging to production.

This workflow is, I assume, super common and it would be helpful if the docs had a section describing how that branching workflow should be created.

Also, I don't think this is possible to setup while on a Hobby (free) plan. You have to upgrade to Scaler or above to gain access to two production branches.  I'd like to see Planetscale relax this limit so the approach could be applied to a Hobby plan for initial experimentation and setup, and then allow easy upgrade to the Scaler plan as need dictates (with no branching setup changes required).
